### ISSUE TITLE
Update GoodCollective to support Segmented UBI

### ISSUE THREAD

**Issue by decentralauren** [GoodDollar/GoodCollective#192]
## Business Description
The GoodCollective dapp is intended to support multiple types of collectives beyond the Direct Payment collectives built for the MVP. We will expand first to support segmented UBI pools, in which beneficiaries are eligible to claim a specified amount of UBI based on decisions made by the pool's sponsor. 

## Detailed Description of Functionality
These pools should share the following functionality with Direct Payment pools:
- There are both a list of beneficiaries and of donors, which display in the GoodCollective dapp
- There is a set amount to which beneficiaries are entitled
- Anyone can donate to the pool either through the GoodCollective interface or by sending supported currencies to the contract address

These pools are distinct from Direct Payment pools in the following ways:
- Only eligible wallets may be beneficiaries (that have either been greenlisted or whose GoodID segmentation match the parameters of the pool)
- Beneficiaries can claim on a regular basis (frequency set by the pool administrator). 
- Using the claim function is the only way beneficiaries can claim (versus Direct Payment pools, in which payments are automatically sent to benificary based on contract activity)
- Should beneficiaries not claim, the funds remain in the pool for future claims.
- Beneficiary eligibility can change, enabling or removing access to the pool

## Technical Implementation
*_Outline the technical approach for building the feature. This section is typically filled out by the development team or can contain suggestions._ 
(dev team: @johnsmith-gooddollar @sirpy @L03TJ3) 
- [ ] (What api's / sdk's could possibly be used)

## Design Reference
[Figma File
](https://www.figma.com/design/ihw1PxBvLxacTHnN2aj4lC/3.-Product?node-id=11695-69772&t=4GrTI0cp6ekNxub0-1)
<img width="241" alt="Screenshot 2024-06-27 at 3 27 33 PM" src="https://github.com/GoodDollar/GoodCollective/assets/149098646/142bd7c8-c01c-4598-b02b-56f5d03a334f">

 
## Acceptance Criteria
*Specify criteria that will be used to determine if the feature meets the requirements and functions correctly.*
- [ ] (Should include a list of testing points for QA how to verify design/functionality)

**Comment by L03TJ3** [comment]
@decentralauren When can the UI update be expected? is there already something on figma?

**Comment by decentralauren** [comment]
@L03TJ3 yes! I haven't reviewed in a long time (5 months) but we started it. Update the initial ticket w the figma.

**Comment by decentralauren** [comment]
@L03TJ3 ticket for updating the GoodCollective pools to include Segmented UBI. Prob need a way to determine pool type based on certain values in the contract? Please begin in this sprint once other priroity items are cleared out :)

**Comment by L03TJ3** [comment]
All done closed
